server/service/integration_mdm_test.go

```diff
@@ -2247,6 +2247,146 @@ func (s *integrationMDMTestSuite) TestMDMDiskEncryptionSettingBackwardsCompat()
+		errMsg = extractServerErrorText(res.Body)
+		require.Contains(t, errMsg, "Couldn't edit enable_disk_encryption. Neither macOS MDM nor Windows is turned on. Visit https://fleetdm.com/docs/using-fleet to learn how to turn on MDM.")
+
+		// try to edit the existing team using specs
+		teamSpecs = map[string]any{
+			"specs": []any{
+				map[string]any{
+					"name": teamName,
+					"mdm": map[string]any{
+						"enable_disk_encryption": true,
+					},
+				},
+			},
+		}
+		res = s.Do("POST", "/api/latest/fleet/spec/teams", teamSpecs, http.StatusUnprocessableEntity)
+		errMsg = extractServerErrorText(res.Body)
+		require.Contains(t, errMsg, "Couldn't edit enable_disk_encryption. Neither macOS MDM nor Windows is turned on. Visit https://fleetdm.com/docs/using-fleet to learn how to turn on MDM.")
+	}
+
+	checkConfigSetSucceeds := func() {
+		res := s.Do("PATCH", "/api/latest/fleet/config", json.RawMessage(`{
+		"mdm": { "enable_disk_encryption": true }
+  }`), http.StatusOK)
+		errMsg := extractServerErrorText(res.Body)
+		require.Empty(t, errMsg)
+
+		// try to create a new team using specs
+		teamSpecs := map[string]any{
+			"specs": []any{
+				map[string]any{
+					"name": teamName + uuid.NewString(),
+					"mdm": map[string]any{
+						"enable_disk_encryption": true,
+					},
+				},
+			},
+		}
+		res = s.Do("POST", "/api/latest/fleet/spec/teams", teamSpecs, http.StatusOK)
+		errMsg = extractServerErrorText(res.Body)
+		require.Empty(t, errMsg)
+
+		// edit the existing team using specs
+		teamSpecs = map[string]any{
+			"specs": []any{
+				map[string]any{
+					"name": teamName,
+					"mdm": map[string]any{
+						"enable_disk_encryption": true,
+					},
+				},
+			},
+		}
+		res = s.Do("POST", "/api/latest/fleet/spec/teams", teamSpecs, http.StatusOK)
+		errMsg = extractServerErrorText(res.Body)
+		require.Empty(t, errMsg)
+
+		// always try to set the value to `false` so we start fresh
+		s.Do("PATCH", "/api/latest/fleet/config", json.RawMessage(`{
+		"mdm": { "enable_disk_encryption": false }
+  }`), http.StatusOK)
+		teamSpecs = map[string]any{
+			"specs": []any{
+				map[string]any{
+					"name": teamName,
+					"mdm": map[string]any{
+						"enable_disk_encryption": false,
+					},
+				},
+			},
+		}
+		s.Do("POST", "/api/latest/fleet/spec/teams", teamSpecs, http.StatusOK)
+	}
+
+	// 1. disable both windows and mac mdm
+	// 2. turn off windows feature flag
+	// we should get an error
+	setMDMEnabled(false, false)
+	t.Setenv("FLEET_DEV_MDM_ENABLED", "0")
+	checkConfigSetErrors()
+
+	// turn on windows feature flag
+	// we should get an error
+	t.Setenv("FLEET_DEV_MDM_ENABLED", "1")
+	checkConfigSetErrors()
+
+	// enable windows mdm, no errors
+	setMDMEnabled(false, true)
+	checkConfigSetSucceeds()
+
+	// enable mac mdm, no errors
+	setMDMEnabled(true, true)
+	checkConfigSetSucceeds()
+
+	// only macos mdm enabled, no errors
+	setMDMEnabled(true, false)
```

Comment:
Just to be sure, are those settings properly reset as part of the common teardown of a test (to make sure subsequent tests work as expected)?

Reply:
I don't think they are. Thank you! will update